test/traversals/dfs.jl

```diff
@@ -26,9 +27,11 @@
 
   @testset "is_cyclic" begin
     for g in testgraphs(path_graph(2))
-      @test @inferred(is_cyclic(g))
+      @test !@inferred(is_cyclic(g))
       @test @inferred(!is_cyclic(zero(g)))
     end
+    @test @inferred(is_cyclic((gcyclic)))
+    @test @inferred(!is_cyclic(gtree))
```

Comment:
I think there are a few more test-cases we have to cover. First, its often good to use graphs with different eltypes (here you use only `Int`)  as a lot of errors happen when that is not the case. 

Apart from that, it is often good to have these edge cases in your test:
- A graph with zero vertices
- A graph with isolated vertices
- A graph with multiple components (this is already handled by your test case)
- A graph with self-loops (we definitely need this test case here)